### PR TITLE
version 1 custom item stats

### PR DESCRIPTION
**Pull request by TesterWoWDev**
**Changes proposed**:

- 
- 
- 

**Target branch(es)**: 335/6x

**Issues addressed**: Fixes #

**Tests performed**: (Does it build, tested in-game, etc)

**Known issues and TODO list**:

- [ ] 
- [ ]

### CODE REVIEW

**Comment by TesterWoWDev** [comment]
also updated this 1

File: sql/custom/characters/custom_item_template.sql [hunk 1]
```diff
@@ -0,0 +1 @@
...
  `ItemLimitCategory` smallint(6) NOT NULL DEFAULT 0,
  `HolidayId` int(11) UNSIGNED NOT NULL DEFAULT 0,
  `ScriptName` varchar(64) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL DEFAULT '',
  `DisenchantID` mediumint(8) UNSIGNED NOT NULL DEFAULT 0,
  `FoodType` tinyint(3) UNSIGNED NOT NULL DEFAULT 0,
  `minMoneyLoot` int(10) UNSIGNED NOT NULL DEFAULT 0,
  `maxMoneyLoot` int(10) UNSIGNED NOT NULL DEFAULT 0,
  `flagsCustom` int(10) UNSIGNED NOT NULL DEFAULT 0,
  PRIMARY KEY (`entry`) USING BTREE
);
```
Comment (tswow-admin) [inline sql/custom/characters/custom_item_template.sql]: Split this up into lines, it's very hard to read like this.

File: src/server/game/Entities/Creature/CreatureOutfit.h [hunk 2]
```diff
@@ -56,6 +56,7 @@ class TC_GAME_API CreatureOutfit
     uint8 GetGender() const { return gender; }
     uint8 GetRace() const { return race; }
     uint32 GetDisplayId() const { return displayId; }
+    void SetDisplayId(uint32 displayID) { displayId = displayID; }
```
Comment (tswow-admin) [inline src/server/game/Entities/Creature/CreatureOutfit.h:59]: Unrelated to this pr, but these should both be changed to GetDisplayID/SetDisplayID as per tswow livescripting conventions.

File: src/server/game/Entities/Item/ItemTemplate.cpp [hunk 3]
```diff
@@ -293,3 +293,17 @@ WorldPacket ItemTemplate::BuildQueryData(LocaleConstant loc) const
     response.ShrinkToFit();
     return response.Move();
 }
+
+// @tswow-begin
+WorldPacket* ItemTemplate::GetQueryData(uint32 index)
+{
+    // todo: hacky, but we need to call this from places thinking these are const.
+    //       It should be safe from these locations.
```
Comment (tswow-admin) [inline src/server/game/Entities/Item/ItemTemplate.cpp]: What does this mean?

File: src/server/game/Globals/ObjectMgr.cpp [hunk 4]
```diff
@@ -3532,6 +3532,336 @@ void ObjectMgr::LoadItemTemplates()
...
+        //                                            126                 127                     128            129            130            131         132         133
+        "GemProperties, RequiredDisenchantSkill, ArmorDamageModifier, duration, ItemLimitCategory, HolidayId, ScriptName, DisenchantID, "
+        //                                           134        135            136
+        "FoodType, minMoneyLoot, maxMoneyLoot, flagsCustom FROM item_template WHERE entry = %s", entry);
+
+    //QueryResult resultNum = WorldDatabase.Query("SELECT entry FROM item_template");
+    //_itemTemplateStore.reserve(resultNum->GetRowCount());
+    //possibly restore old itemTemplateStore?
+
+    while (result->NextRow())
```
Comment (tswow-admin) [inline src/server/game/Globals/ObjectMgr.cpp]: This is wrong. You will skip the first row this way, use a do-while loop

File: src/server/game/Globals/ObjectMgr.h [hunk 5]
```diff
@@ -1244,6 +1244,12 @@ class TC_GAME_API ObjectMgr
         void LoadSpawnGroups();
         void LoadInstanceSpawnGroups();
         void LoadItemTemplates();
+        // @tswow-begin
+        void LoadSingleItemTemplate(std::string entry);
```
Comment (tswow-admin) [inline src/server/game/Globals/ObjectMgr.h]: It looks like this function isn't being used by anything, maybe that's why it was left with an incorrect loop while the other one was correct.

It's also weird to use a string as an argument, if you want to keep it that should be a uint32.

**Review by tswow-admin — CHANGES_REQUESTED** [review]
Only minor points

File: src/server/game/Entities/Creature/CreatureOutfit.h (hunk 2 above)
Comment (ihm-tswow) [reply src/server/game/Entities/Creature/CreatureOutfit.h:59]: We can change this one to SetDisplayID, deprecate GetDisplayId and then add GetDisplayID

File: src/server/game/Entities/Item/ItemTemplate.cpp (hunk 3 above)
Comment (ihm-tswow) [reply src/server/game/Entities/Item/ItemTemplate.cpp]: nothing, it's a leftover

File: src/server/game/Globals/ObjectMgr.cpp (hunk 4 above)
Comment (ihm-tswow) [reply src/server/game/Globals/ObjectMgr.cpp]: I removed this function

File: src/server/game/Entities/Creature/CreatureOutfit.h (hunk 2 above)
Comment (ihm-tswow) [reply src/server/game/Entities/Creature/CreatureOutfit.h:59]: lol, this isn't even the tswow api, you tricked me!

File: sql/custom/characters/custom_item_template.sql (hunk 1 above)
Comment (ihm-tswow) [reply sql/custom/characters/custom_item_template.sql]: Those look like crlf line endings that seem to have messed up for some reason, i fixed it

**Review by tswow-admin — APPROVED** [review]

**Pull request merged**